### PR TITLE
Add suspect commit type to CommitSerialier

### PR DESCRIPTION
**Pull request by snigdhas**
Adds `via commit in release` or `via SCM integration` to the serialized commit so we can indicate the source of suspect commits in the UI. 

[WOR-2717](https://getsentry.atlassian.net/browse/WOR-2717)

[WOR-2717]: https://getsentry.atlassian.net/browse/WOR-2717?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/sentry/utils/committers.py [hunk 1]
```diff
@@ -322,7 +322,12 @@ def get_serialized_event_file_committers(
         return [
             {
                 "author": author,
-                "commits": [serialize(commit, serializer=CommitSerializer(exclude=["author"]))],
+                "commits": [
+                    serialize(
+                        commit,
+                        serializer=CommitSerializer(exclude=["author"], type="via SCM integration"),
```
Comment (markstory) [inline src/sentry/utils/committers.py:328]: Are these types intended to behave like an enum or are they text to be presented to end users?

Comment (snigdhas) [reply src/sentry/utils/committers.py:328]: I suppose either would work, but the end goal is to show it to users in the UI.

Comment (markstory) [reply src/sentry/utils/committers.py:328]: Returning an enum value would make it simpler to add translations in the future. If the API endpoint is returning UI text, you're limited in what formatting can be done and we would need to add translations server side (if we ever get around to updating translations).

**Review by markstory — APPROVED** [review]
Looks good to me.

**Pull request merged**